src/udiskslinuxmanager.c

```diff
@@ -1069,6 +1069,106 @@ handle_get_block_devices (UDisksManager         *object,
   return TRUE;  /* returning TRUE means that we handled the method invocation */
 }
 
+static gboolean
+compare_paths (UDisksManager         *object,
+               UDisksBlock           *block,
+               gchar                 *path)
+{
+  const gchar *const *symlinks = NULL;
+
+  if (g_strcmp0 (udisks_block_get_device (block), path) == 0)
+    return TRUE;
+
+  symlinks = udisks_block_get_symlinks (block);
+  if (symlinks != NULL)
+    {
+      for (guint i = 0; symlinks[i] != NULL; i++)
+        if (g_strcmp0 (symlinks[i], path) == 0)
+          return TRUE;
+    }
+
+  return FALSE;
+}
+
+static gboolean
+handle_resolve_device (UDisksManager         *object,
+                       GDBusMethodInvocation *invocation,
+                       GVariant              *arg_devspec,
+                       GVariant              *arg_options)
+{
+
+  gchar *devpath = NULL;
+  gchar *devuuid = NULL;
+  gchar *devlabel = NULL;
+
+  GSList *blocks = NULL;
+  GSList *blocks_p = NULL;
+  guint num_blocks = 0;
+
+  GSList *ret = NULL;
+  GSList *ret_p = NULL;
+  guint num_found = 0;
+  const gchar **ret_paths = NULL;
+
+  gboolean found = FALSE;
+  guint i = 0;
+
+  g_variant_lookup (arg_devspec, "path", "s", &devpath);
+  g_variant_lookup (arg_devspec, "uuid", "s", &devuuid);
+  g_variant_lookup (arg_devspec, "label", "s", &devlabel);
+
+  blocks = get_block_objects (object, &num_blocks);
+
+  for (blocks_p = blocks; blocks_p != NULL; blocks_p = blocks_p->next)
+    {
+      if (devpath != NULL)
+        {
+          if (compare_paths (object, blocks_p->data, devpath))
+            found = TRUE;
+          else
+            found = FALSE;
+        }
```

Comment:
``found = compare_paths (object, blocks_p->data, devpath))``

And the same applies to all those ``if``s below.